clients/src/main/java/org/apache/kafka/common/metrics/QuotaViolationException.java

```diff
@@ -30,11 +30,7 @@
     private final double bound;
 
     public QuotaViolationException(MetricName metricName, double value, double bound) {
-        super(String.format(
-                "'%s' violated quota. Actual: %f, Threshold: %f",
-                metricName,
-                value,
-                bound));
+        super(metricName + " violated quota.");
```

Comment:
I think we want to make the string computation lazy, but not lose info. We could do that by overriding `toString`.

Reply:
Great idea. I'll change that.

Reply:
I've fixed this. toString for an example violation before and after both returned:
```
org.apache.kafka.common.metrics.QuotaViolationException: 'MetricName [name=x, group=y, description=z, tags={}]' violated quota. Actual: 3.300000, Threshold: 3.300000                                   
```

Reply:
Can we remove the allocation altogether in the constructor?

Reply:
Good point, now that toString is overridden I had actually just made that change and hadn't pushed it

Reply:
Done